humility-net-core/src/lib.rs

```diff
@@ -314,7 +312,13 @@ impl Core for NetCore {
 
     fn send(&self, buf: &[u8], target: NetAgent) -> Result<usize> {
         match target {
-            NetAgent::UdpRpc => self.udprpc_socket.send(buf),
+            NetAgent::UdpRpc => {
+                if let Some(d) = self.udprpc_socket.as_ref() {
+                    d.send(buf)
+                } else {
+                    bail!("no `udprpc` socket");
```

Comment:
Not sure if this is a message that an errant user could hit, but do we want to prompt around the image type as in the other error messages?

Reply:
I _think_ this isn't hittable by users, because we check during `HiffyContext` construction whether the `udprpc` task exists and bail out otherwise.